docs/contributing.md

```diff
@@ -219,64 +219,67 @@ multi-threaded. It should use all the CPU cores without any configuration.
 
 ## Previewing the HTML documentation
 
-The documentation for `jj` is automatically published to the website at
+The documentation for `jj` is automatically published online at
 <https://martinvonz.github.io/jj/>.
 
-When editing documentation, we'd appreciate it if you checked that the
-result will look as expected when published to the website.
+When editing documentation, you should check your changes locally — especially
+if you are adding a new page, or doing a major rewrite.
 
-### Setting up the prerequisites
+The only thing you need is [`uv`][uv], a Python project manager written in Rust.
+It will fetch the right Python version and the dependencies. Install it like so:
 
-To build the website, you must have Python and `poetry 1.8+` installed (the
-latest version is recommended). It is easiest to install `poetry` via `pipx`, as
-explained in the [Poetry installation instructions]. A few helpful points from
-the instructions: `pipx` can often be installed from your distribution, e.g.
-`sudo apt install pipx`; this will usually also install Python for you if
-necessary. Any version of `pipx` will do. If you are installing `pipx` manually,
-you may first need to follow the [Python installation instructions].
+[uv]: https://docs.astral.sh/uv/
 
-[Python installation instructions]: https://docs.python.org/3/using/index.html
-[Poetry installation instructions]: https://python-poetry.org/docs/#installation
+=== "macOS/Linux"
```

Comment:
Yeah, this tradeoff comes up often :( Not just with github, but also when people read docs in the editor or use built-in preview (VSCode and friends don’t know about mkdocs extensions).

Do people read jj docs on GitHub in practice tho, or is it more of a theoretical concern?

Reply:
I think people will read this specific doc from GitHub. OTOH, one way around that would be to add a file named CONTRIBUTING.md to the top-level of `jj` repo and have a link to the website from there.

Perhaps I'll send a PR for that.

Reply:
On second thought, this would make the link in the screenshot below work worse, and it'd duplicated text in the README. So, I'm doing a simpler PR for now.

[GitHub's instructions](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) make it seem like this is impossible to customize.

I'm thinking that we could instead add a banner to the existing file that appears only on GitHub and points to the website.

<img width="1284" alt="SCR-20241115-nxcu" src="https://github.com/user-attachments/assets/b499d8cb-6802-4516-b274-62c259686ed4">

Reply:
If GitHub allows `<div class="hide-on-website">`, we could add some CSS to the mkdocs-material theme to hide that class. That's the best idea I have at the moment.